lib/variable/inverse.cpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+#include "scipp/variable/inverse.h"
+
+#include "scipp/common/overloaded.h"
+#include "scipp/core/element/arg_list.h"
+#include "scipp/core/spatial_transforms.h"
+#include "scipp/core/transform_common.h"
+
+#include "scipp/variable/transform.h"
+
+namespace scipp::variable {
+namespace element {
+constexpr auto inverse =
+    overloaded{core::element::arg_list<Eigen::Matrix3d, Eigen::Affine3d,
+                                       core::Translation, core::Quaternion>,
+               core::transform_flags::expect_no_variance_arg<0>,
+               core::transform_flags::expect_no_variance_arg<1>,
+               [](const auto &transform) { return transform.inverse(); },
+               [](const units::Unit &u) {
+                 // This is not correct for linear transforms (Eigen::Matrix3d).
+                 // But that case is handled separately by the caller.
+                 return u;
```

Comment:
I would consider translations the "special" case, and return `1/u` here, handling translations in the caller?

Reply:
Affine transforms also have a unit. And it applies only to the translational part. So we would have to special case that as well.